src/job/runner.go

```diff
@@ -105,14 +104,14 @@ func (r *Runner) Run(ctx context.Context, logger *zap.Logger) {
 				cancel()
 			}
 
+			r.reporter.ClearData() // clear info about previous targets
```

Comment:
The main reason reporter was recreated with each new config is so that jobs that haven't been killed right away wouldn't have a chance to report their metrics to the new iteration. This approach may bring that bug back. Can we instead split the reporter interface into actual reporting/logging part and storage part? Then reporter could have a single method for writing summary that accepts a storage and we would be able to recreate storage on each new config

Reply:
Actually, it's already split - `Reporter` interface with two implementations and `ReporterData` (which I'll probably rename to `Metrics` or something) struct with all info. Yep, it would be even easier not to store it inside I guess.. I'll look into it

Reply:
Split it, now Metrics are about holding combined stats and Reporter is about reporting. The bug should be prevented now, I create a new storage on a new config.